tests/kernel/mutex/mutex_null_ptr_deref/src/main.c

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright (c) 2022 SeediqQ from Issue 48056
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/ztest.h>
+
+#define TIMEOUT_MS 100
+#define STACKSZ	   8192
+
+static struct k_mutex mutex;
+static struct k_thread thread;
+
+K_THREAD_STACK_DEFINE(stack, STACKSZ);
+
+static void test_thread(void *p1, void *p2, void *p3)
+{
+	ARG_UNUSED(p1);
+	ARG_UNUSED(p2);
+	ARG_UNUSED(p3);
+
+	k_mutex_lock(&mutex, K_MSEC(TIMEOUT_MS));
+
+	k_mutex_unlock(&mutex);
+}
+
+ZTEST(mutex_timeout_null_ptr_deref, test_mutex_timeout_error)
+{
```

Comment:
There should be a restriction that this test only be executed on a uniprocessor system. Otherwise on an MP system, we may be not even validating PR #47906 as both threads could be executing simultaneously on different cores.

Reply:
what is the best way to achieve that today? I recall ztest 1cpu existed with the old framework.

Reply:
`CONFIG_NUM_CPUS=1`